EXTI/Core/Src/functions.c

```diff
@@ -0,0 +1,55 @@
+/*
+ * functions.c
+ *
+ *  Created on: Mar 5, 2020
+ *      Author: rodrigocelmer
+ */
+#include "stm32l053xx.h"
+#include "macros.h"
+
+extern int	user_bt_count,	//counter to debounce the button
+			flag_EXTI,		//flag for EXTI interrupt
+			flag_TIM6,		//flag for TIM6 interrupt
+			flag_frequency;	//flag to indicate if we are increasing or decreasing the frequency
+
+void Blink_Red_Led (void)
+{
+	flag_TIM6 = 0;		//clears TIM6 flag
+	TOGGLE_RED_LED;		//toggle the red led every 100 ms
+}
+
+void Change_Frequency (void)
+{
+	user_bt_count++;							//counts up to 50 for debouncing
+	if(user_bt_count==50)
+	{
+		if((TIM6->ARR > 19)&& flag_frequency == UP)
+		{
+			TIM6->ARR = TIM6->ARR - 10;
```

Comment:
The operation `TIM6->ARR = TIM6->ARR - 10;` works great, but I suggest you to write it in a easier way, like this:
`TIM6->ARR -= 10;`
Both ways results in the same thing, but the second one is easier to read.